src/components/ha-circular-progress.ts

```diff
@@ -1,54 +1,39 @@
-import { CircularProgress } from "@material/mwc-circular-progress";
-import { CSSResultGroup, css } from "lit";
+import "element-internals-polyfill";
+import { MdCircularProgress } from "@material/web/progress/circular-progress";
+import { CSSResult, css } from "lit";
 import { customElement, property } from "lit/decorators";
 
 @customElement("ha-circular-progress")
-// @ts-ignore
-export class HaCircularProgress extends CircularProgress {
-  @property({ type: Boolean })
-  public active = false;
+export class HaCircularProgress extends MdCircularProgress {
+  @property({ attribute: "aria-label", type: String }) public ariaLabel =
+    "Loading";
 
-  @property()
-  public alt = "Loading";
+  @property() public size: "tiny" | "small" | "medium" | "large" = "medium";
 
-  @property()
-  public size: "tiny" | "small" | "medium" | "large" = "medium";
+  public firstUpdated(changedProperties) {
+    super.firstUpdated(changedProperties);
 
-  // @ts-ignore
-  public set density(_) {
-    // just a dummy
-  }
-
-  public get density() {
     switch (this.size) {
       case "tiny":
-        return -8;
+        this.style.setProperty("--md-circular-progress-size", "16px");
+        break;
       case "small":
-        return -5;
-      case "medium":
-        return 0;
+        this.style.setProperty("--md-circular-progress-size", "28px");
+        break;
+      // medium is default size
```

Comment:
This should still be set (or unset), if it is first set to small and then changed to medium it will not reset.